### DISCUSSION TITLE
NVR harware with 2x 3"5 disks? Or do we only need machines with SSD

### DISCUSSION THREAD

**Discussion by benoitc** [blakeblackshear/frigate#6398] [General]
Looking at the hardware documentation I only see small machines with SSD but no machines allowing to record for a long time. How people do afaik. Do yo store on a NAS? What kind of machine equivalent to https://www.synology.com/fr-fr/products/DVA ?

**Comment by NickM-27** [comment]
there are many different approaches based on how much recording you want to maintain. Frigate is run in docker so you can mount whatever type of storage you want to the recordings directory.

personally, I have an Unraid NAS that I run frigate on directly. Frigate recordings to SSDs and then when the SSD is full the recordings are moved to an HDD until they are eventually expired and deleted. Frigate sees these as one large volume and all the management is handled by Unraid

**Comment by LaurenceGough** [comment]
It's a great question as CCTV IMO is not well suited to SSDs, I mean it's fine if you don't mind the wear on them and it's probably not really an issue I just don't like the idea.

I am currently using a NAS with an option for one or two 3.5" HDDs/SSDs. It's a very old NAS now and it does have a performance impact, I think Frigate is quite "chatty" in IOs, I say that completely unfounded but when I transfer files manually back and forth to the NAS I get a high speed, but IO probably isn't great on this NAS. However, saying that with the very latest Frigate dev builds it is damn instant loading everything with the h265 recordings.  h264 larger res recordings take a bit longer due to my old NAS.

I've now purchased a mini PC and it has a SATA slot for a 2.5" HDD or SSD.  2.5" HDDs use less power (and therefore shouldn't need active cooling like 3.5" HDD does) and seem online to be just as reliable. They don't make CCTV specific 2.5" drives, but there is one dedicated to 24/7 recording and video streaming. The WD AV-25 HDD (1TB max). I have ordered one to save power from having to run a NAS and to speed up my Frigate system, but it has yet to arrive :(. 

If you need RAID for redundancy you best look for a multi slot NAS or multi SATA device with RAID support, some more powerful ones can run Frigate from the NAS too at a cost (the NAS's tend to be quite expensive that can do this).

I run Frigate etc from the internal SSD.

**Comment by blakeblackshear** [comment]
I run the OS off the SSD and write recordings to an HDD (either external USB3 or internal).
